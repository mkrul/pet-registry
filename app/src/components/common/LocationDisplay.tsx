import React from "react";

interface LocationDisplayProps {
  area?: string | null;
  state?: string | null;
  country?: string | null;
}

const LocationDisplay: React.FC<LocationDisplayProps> = ({ area, state, country }) => {
  const locationParts = [area, state, country].filter(Boolean);

  if (locationParts.length === 0) {
    return null;
  }

  return <p className="text-gray-500 mt-2 mb-4">{locationParts.join(", ")}</p>;
};

export default LocationDisplay;
