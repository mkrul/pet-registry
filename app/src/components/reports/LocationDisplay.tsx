import React from "react";

interface LocationDisplayProps {
  area: string;
  state: string;
  country: string;
}

export const LocationDisplay: React.FC<LocationDisplayProps> = ({ area, state, country }) => {
  return (
    <div className="mt-4">
      <h2 className="text-lg font-medium text-gray-900 mb-2">Location</h2>
      <div className="grid grid-cols-3 gap-4">
        <div>
          <div className="text-md text-gray-500 mb-1">Area</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {area || "No area selected"}
          </div>
        </div>
        <div>
          <div className="text-md text-gray-500 mb-1">State</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {state || "No state selected"}
          </div>
        </div>
        <div>
          <div className="text-md text-gray-500 mb-1">Country</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {country || "No country selected"}
          </div>
        </div>
      </div>
    </div>
  );
};
