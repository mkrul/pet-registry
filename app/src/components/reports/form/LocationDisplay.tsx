import React from "react";
import { LocationDisplayProps } from "../../../types/Report";

export const LocationDisplay: React.FC<LocationDisplayProps> = ({ area, state, country }) => {
  return (
    <div className="mt-4">
      <h2 className="text-lg font-medium text-gray-900 mb-2">Location</h2>
      <div className="grid grid-cols-3 gap-4">
        <div>
          <div className="text-md text-gray-500 mb-1">Area</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {area}
          </div>
        </div>
        <div>
          <div className="text-md text-gray-500 mb-1">State</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {state}
          </div>
        </div>
        <div>
          <div className="text-md text-gray-500 mb-1">Country</div>
          <div className="p-2 bg-gray-50 border border-gray-200 rounded-md text-gray-700 truncate">
            {country}
          </div>
        </div>
      </div>
    </div>
  );
};
