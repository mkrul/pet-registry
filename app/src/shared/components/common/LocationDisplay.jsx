import React from "react";
import { getStateAbbreviation } from "../../utils/stateUtils";
const LocationDisplay = ({
  area,
  state,
  textStyle = "text-gray-500",
  intersection,
  displayTip,
  useStateAbbreviation = false
}) => {
  const locationParts = [area, state, "United States", intersection].filter(Boolean);

  if (locationParts.length === 0) {
    return null;
  }

  const locationString = () => {
    if (area && state) {
      const stateDisplay = useStateAbbreviation ? getStateAbbreviation(state) : state;
      if (intersection && intersection !== "") {
        return <p>Reported missing near {intersection} in {area}, {stateDisplay}</p>;
      }
      return <p>Reported missing in {area}, {stateDisplay}</p>;
    }
    return null;
  };

  return (
    <div className={`${textStyle}`}>
      {locationString()}
    </div>
  );
};

export default LocationDisplay;
