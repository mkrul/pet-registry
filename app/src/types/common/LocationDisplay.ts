export interface LocationDisplayProps {
  area?: string | null;
  state?: string | null;
  country?: string | null;
  intersection?: string | null;
}
