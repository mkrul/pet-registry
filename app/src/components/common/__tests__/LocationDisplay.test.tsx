import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/vitest";
import LocationDisplay from "../LocationDisplay";

describe("LocationDisplay", () => {
  const testCases = [
    {
      name: "renders full location with all props",
      props: { area: "Brooklyn", state: "New York", country: "United States" },
      expected: "Brooklyn, New York, United States"
    },
    {
      name: "renders location with area and country only",
      props: { area: "Paris", country: "France" },
      expected: "Paris, France"
    },
    {
      name: "renders location with state and country only",
      props: { state: "Ohio", country: "United States" },
      expected: "Ohio, United States"
    },
    {
      name: "renders single location part",
      props: { country: "United States" },
      expected: "United States"
    }
  ];

  testCases.forEach(({ name, props, expected }) => {
    it(name, () => {
      render(<LocationDisplay {...props} />);
      const location = screen.getByText(expected);
      expect(location).toBeDefined();
      expect(location).toHaveClass("text-gray-500", "mt-2");
    });
  });

  it("returns null when no location props are provided", () => {
    const { container } = render(<LocationDisplay />);
    expect(container.firstChild).toBeNull();
  });

  it("handles empty string props by filtering them out", () => {
    render(<LocationDisplay area="" state="California" country="" />);
    expect(screen.getByText("California")).toBeDefined();
  });

  it("matches snapshot with full location", () => {
    const { container } = render(
      <LocationDisplay area="Manhattan" state="New York" country="United States" />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with partial location", () => {
    const { container } = render(<LocationDisplay state="Ohio" country="United States" />);
    expect(container).toMatchSnapshot();
  });
});
