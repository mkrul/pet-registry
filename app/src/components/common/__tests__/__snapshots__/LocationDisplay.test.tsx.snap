// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocationDisplay > matches snapshot with full location 1`] = `
<div>
  <p
    class="text-gray-500 mt-2"
  >
    Manhattan, New York, USA
  </p>
</div>
`;

exports[`LocationDisplay > matches snapshot with partial location 1`] = `
<div>
  <p
    class="text-gray-500 mt-2"
  >
    Ontario, Canada
  </p>
</div>
`;
